src/mempool/mempool.rs

```diff
@@ -26,14 +30,93 @@ use crate::{
 };
 
 use super::routing::{MessageRouter, ShardRouter};
+use governor::{Quota, RateLimiter};
+use moka::sync::{Cache, CacheBuilder};
+use std::num::NonZeroU32;
+use thiserror::Error;
 use tracing::{error, warn};
 
+type DirectRateLimiter = RateLimiter<NotKeyed, InMemoryState, QuantaClock>;
+
+#[derive(Error, Debug)]
+pub enum RateLimitsError {
+    #[error("no storage")]
+    NoStorage,
+}
+
+pub struct RateLimitsConfig {
+    pub time_to_idle: Duration,
+    pub max_capacity: usize,
+}
+
+impl Default for RateLimitsConfig {
+    fn default() -> Self {
+        Self {
+            time_to_idle: Duration::from_secs(60 * 2 * 2),
+            // Make time to idle 2x the rate limit window so it's ok if out of sync
+            max_capacity: 10_000,
```

Comment:
This may be too low, this should be at least 100K, but I think we should be even more lenient and set it to 1M. Worst case is 1K msgs/block from unique fids * 1 hour in seconds = 3.6M fids. 

We should emit a stat on how much of this capcaity is used

Reply:
Hmm, this is global, and not shared per shard? Even more important to bump this up.